test-utilities/Package.swift

```diff
@@ -0,0 +1,24 @@
+// swift-tools-version: 5.7
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
+    name: "TestUtilities",
```

Comment:
I'm concerned about this name being a bit to general... does it make sense to maybe call it DatadogTestUtilities?

Reply:
We have this unwritten agreement to prepend targets with `datadog` only if they are user facing. This one is purely internal. Interestingly enough - in this PR it's still quite generic, but next one will be exposing some Datadog specific mocks (so they're usable in other vertical features).